cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -211,20 +211,14 @@ def verify_revert_snapshot(full_test_vm, provider, soft_assert, register_event,
     else:
         snapshot1 = new_snapshot(full_test_vm)
     full_template = getattr(provider.data.templates, 'full_template')
-    # Define parameters of the ssh connection
-    ssh_kwargs = {
-        'hostname': snapshot1.parent_vm.mgmt.ip,
-        'username': credentials[full_template.creds]['username'],
-        'password': credentials[full_template.creds]['password']
-    }
-    ssh_client = SSHClient(**ssh_kwargs)
-    # We need to wait for ssh to become available on the vm, it can take a while. Without
-    # this wait, the ssh command would fail with 'port 22 not available' error.
-    # Easiest way to solve this is just mask the exception with 'handle_exception = True'
-    # and wait for successful completition of the ssh command.
-    # The 'fail_func' ensures we close the connection that failed with exception.
-    # Without this, the connection would hang there and wait_for would fail with timeout.
-    wait_for(lambda: ssh_client.run_command('touch snapshot1.txt').success, num_sec=400,
+    # We need to wait for ssh to become available on the vm, it can take a while.
+    # connect_ssh will iterate over "all_ips" on the VM and return a client when it can connect
+    creds = {
+        'principal': credentials[full_template.creds]['username'],
+        'password': credentials[full_template.creds]['password']}
+    with connect_ssh(full_test_vm, creds) as ssh_client:
+        yield ssh_client
+    wait_for(lambda: ssh_client.run_command('touch snapshot1.txt').success, num_sec=200,
```

Comment:
I think you should *not* yield here, but rather put the rest of the code until the line 265 under the `with` statement and remove the `ssh_client.close()` as the `with` will do that for you